File: tasks/lectures/hongik/python/midterm/4.py
n = float(input('1보다 작고 0보다 큰 소수를 입력하세요 : '))

def unit_fraction(frac: float):
    loop_cnt = 2
    b, a = 0, 0
    while True:
        b, a = 1/(loop_cnt-1), 1/loop_cnt # Before, After
        if frac < b and frac >= a:
            break
        loop_cnt += 1
    near = min(b - frac, frac - a)
    if near == b - frac:
        return b, loop_cnt - 1
    else:
        return a, loop_cnt

res, lower = unit_fraction(n)
print('가장 가까운 단위 분수는 1/{}이며 이 값은 {}'.format(lower, res))

File: tasks/lectures/jnu/java-ai-basic-jho-2022/task-2/java/4.java
public class Main {
  public static void main(String[] args) {
    // 문제 제시 값
    int n[][] = {{1},{1,2,3},{1},{1,2,3,4},{1,2}};

    // 배열을 행렬로 생각했을 때
    // 첫번째 for문으로 행 조회
    for (int i = 0; i < n.length; i++) {
      // 두번째 for문으로 열 조회
      for (int j = 0; j < n[i].length; j++) {
        // (i, j)번 요소 조회
        System.out.print(String.format("%d ", n[i][j]));
      }
      // 하나의 열을 모두 조회하면 개행
      System.out.println();
    }
  }
}